redfish-core/lib/managers.hpp

```diff
@@ -1962,6 +1962,55 @@ inline void
     });
 }
 
+inline void getBMCState(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
+{
+    aResp->res.jsonValue["PowerState"] = "On";
+    sdbusplus::asio::getProperty<std::string>(
+        *crow::connections::systemBus, "xyz.openbmc_project.State.BMC",
+        "/xyz/openbmc_project/state/bmc0", "xyz.openbmc_project.State.BMC",
+        "CurrentBMCState",
+        [aResp](const boost::system::error_code ec,
+                const std::string& bmcState) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error reading CurrentBmcState");
+            aResp->res.jsonValue["Status"]["State"] = "Enabled";
```

Comment:
3 of these Enabled/OK could rewrite as  

```
    aResp->res.jsonValue["PowerState"] = "On";
    aResp->res.jsonValue["Status"]["State"] = "Enabled";
    aResp->res.jsonValue["Status"]["Health"] = "OK";
   
    sdbusplus::asio::getProperty<std::string>(
        *crow::connections::systemBus, "xyz.openbmc_project.State.BMC",
        "/xyz/openbmc_project/state/bmc0", "xyz.openbmc_project.State.BMC",
        "CurrentBMCState",
        [aResp](const boost::system::error_code ec,
                const std::string& bmcState) {
       if (ec)
        {
            BMCWEB_LOG_DEBUG("DBUS response error reading CurrentBmcState");
            // Just use the default
            return;
        }

       // .BMC.BMCState.Ready is just the default
       
        if (bmcState == "xyz.openbmc_project.State.BMC.BMCState."
                             "Quiesced")
        {
            aResp->res.jsonValue["Status"]["State"] = "Quiesced";
            aResp->res.jsonValue["Status"]["Health"] = "Critical";
        }
        else if (bmcState == "xyz.openbmc_project.State.BMC.BMCState."
                             "NotReady")
        {
            aResp->res.jsonValue["Status"]["State"] = "Starting";
        }
        else if (bmcState == "xyz.openbmc_project.State.BMC.BMCState."
                             "UpdateInProgress")
        {
            aResp->res.jsonValue["Status"]["State"] = "Updating";
        }      
        else
        {
            BMCWEB_LOG_DEBUG("Unsupported D-Bus CurrentBMCState: {}", bmcState);
            // Just use the default 
        }
 ```

Reply:
I like a bit more code that's a bit more obvious and readable, and also code that's been tested (1020-1060) so lets leave as-is.